code/mob/living/silicon/ai/holograms.dm

```diff
@@ -1,26 +1,109 @@
 //AI HOLOGRAMS
-/mob/living/silicon/ai/proc/create_hologram()
-	if (!deployed_to_eyecam)
-		boutput(src, "Deploy to an AI Eye first to create a hologram.")
-		return
-	if (length(holograms)>= max_holograms)
-		boutput(eyecam, "Not enough RAM to project more holograms. Delete others to make room.")
-		return
-	//select from list of icon_state in holograms.dmi
-	var/input = input("Select a type of hologram to display.","Hologram") as null|anything in list("caution","o2","beepsky","up_arrow","down_arrow","left_arrow","right_arrow","happy_face","neutral_face","sad_face","angry_face")
-	if (input)
+/mob/living/silicon/ai
+	contextLayout = new /datum/contextLayout/experimentalcircle(36)
+	var/list/hologramTypes = list(
```

Comment:
why not use children/concrete typesof(/datum/contextAction/ai_hologram) and get the holo_types, you're needlessly repeating

Reply:
You're right, didn't think of that. Also applied this to `hologramContextActions`.